src/json-formatter.js

```diff
@@ -92,25 +134,45 @@ angular.module('jsonFormatter', ['RecursionHelper'])
     };
 
     scope.parseValue = function (value){
-      scope.type = getType(scope.json);
-      if (scope.type === 'null') {
-        return 'null';
-      }
-      if (scope.type === 'undefined') {
-        return 'undefined';
-      }
-      if (scope.type === 'string') {
-        value = '"' + escapeString(value) + '"';
-      }
-      if (scope.type === 'function'){
+      return parseValue(scope.json, value);
+    };
 
-        // Remove content of the function
-        return scope.json.toString()
-          .replace(/[\r\n]/g, '')
-          .replace(/\{.*\}/, '') + '{ ... }';
+    scope.showThumbnail = function () {
+      return !!scope.thumbnail;
+    };
+
+    scope.getThumbnail = function () {
+      if (scope.isArray()) {
+        //
+        // if array length is 256, greater then 100
+        // show "Array[256]"
+        if (scope.json.length > 100) {
+          return 'Array[' + scope.json.length + ']';
+        } else {
+          return '[' + scope.json.map(parseThumbnail).join(', ') + ']';
+        }
+      } else {
+
+        var keys = scope.getKeys();
+        //
+        // the first five keys (like Chrome Developer Tool)
+        var narrowKeys = keys.filter(function (e, i) {
+          return i < 5;
+        });
+
+        //
+        // json value schematic information
+        var kvs = narrowKeys
+          .map(function (key) { return [key, scope.json[key]]; }) // javascript array not have zip function %>_<%
+          .map(function (e) { return '' + e[0] + ':' + parseThumbnail(e[1]); });
+
+        //
+        // if keys count greater then 5
+        // then show ellipsis
+        var ellipsis = keys.length >= 5 ? "..." : '';
```

Comment:
1. greater or greater and equal?
2. Use ellipsis unicode character instead of dotdotdot